File: farmos_flutter/lib/models/plot.dart
class Plot {
  final int id;
  final String name;
  final String soilType;
  final double areaM2;
  final double areaAcres;
  final String polygonPoints;
  final String? digitizedDiagram;
  final Map<String, dynamic>? farm;

  Plot({
    required this.id,
    required this.name,
    required this.soilType,
    required this.areaM2,
    required this.areaAcres,
    required this.polygonPoints,
    this.digitizedDiagram,
    this.farm,
  });

  factory Plot.fromJson(Map<String, dynamic> json) {
    return Plot(
      id: json['id'],
      name: json['name'],
      soilType: json['soilType'] ?? '',
      areaM2: (json['areaM2'] as num?)?.toDouble() ?? 0,
      areaAcres: (json['areaAcres'] as num?)?.toDouble() ?? 0,
      polygonPoints: json['polygonPoints'] ?? '[]',
      digitizedDiagram: json['digitizedDiagram'],
      farm: json['farm'],
    );
  }
}

File: farmos/src/main/java/com/farmos/farmos/model/Plot.java
package com.farmos.farmos.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "plots")
public class Plot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String soilType;
    private Double areaM2;
    private Double areaAcres;

    // stores the polygon points as a JSON string
    // e.g. "[{x:100,y:50},{x:200,y:50},{x:150,y:120}]"
    // @Column(columnDefinition) tells PostgreSQL to use TEXT type
    // because the string can be very long
    @Column(columnDefinition = "TEXT")
    private String polygonPoints;

    // stores digitizer metadata as JSON, including the source sketch image and overlay data
    @Column(columnDefinition = "TEXT")
    private String digitizedDiagram;

    // links this plot to a farm
    @ManyToOne
    @JoinColumn(name = "farm_id")
    private Farm farm;
}
